src/main/java/com/example/Restaurant_Management_System_REST_API/model/entity/Customer.java

```diff
@@ -0,0 +1,24 @@
+package com.example.Restaurant_Management_System_REST_API.model.entity;
+
+import com.example.Restaurant_Management_System_REST_API.model.ContactDetails;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.time.LocalDateTime;
+
+@Entity
+@Getter
+@Setter
+public class Customer extends ContactDetails {
```

Comment:
I'm not sure about this structure where Customer extends ContactDetails. This is somehow a little bit counterintuitive to me why is Customer part of ContactDetails and not the other way around? Why is even one extends another?

Reply:
In this context, ContactDetails is an abstract class that holds common information (like name, street, city, postalCode, telephoneNumber, emailAddress) that could be applicable to various entities (like Customer, Supplier, Employee, etc.). So, when Customer extends ContactDetails, it’s inheriting these common properties, indicating that a Customer has these contact details.

If you find this structure confusing, an alternative approach could be to use composition over inheritance. Instead of having Customer extend ContactDetails, I could include ContactDetails as a field within the Customer class.

tell me what you think.

Reply:
That's right, I think inheritance here in this context is not fully correct and/or clear. The other option you described here is more convenient or at last more clear to my with its purpose.

Reply:
And this comment above is my point exactly why requesting review before I was able to answer here is a mistake. That's because I have nothing to review due to the changes that you will now most probably implement. That's FYI to my comment on this matter.

Reply:
@JasinskiSz I didn't get your answer here. please answer

Reply:
@BartoliniAndBorderCollies What's the question I should answer here?

Reply:
Ok sorry your answers haven't appeared earlier to me, now I can see them.
thanks!